### DISCUSSION TITLE
GLSP 2.0 - SModelElement Arguments?

### DISCUSSION THREAD

**Discussion by rsoika** [eclipse-glsp/glsp#1165] [Q&A] [answered]
Hi, 
before glsp version 2.0 I often used code to get/set args of a SModelElement like this:

```javascript
import {
    SModelElement,
    hasArguments
} from '@eclipse-glsp/client';
.....

       let element: SModelElement | undefined;
       ....
        if (element && hasArguments(element)) {
              element.args.....
        }
```

I understand that `SModelElement` is now replaced by `GModelElement`, but the method `hasArguments` is lost?
I did not find any comments about the arguments attribute in the [glsp-client release notes](https://github.com/eclipse-glsp/glsp-client/releases/tag/v2.0.0). 
Is my assumption correct that  `SModelElement` was replaced by `GModelElement`?

**Comment by rsoika** [comment, marked as answer]
...ok I try to answer it by myself: it looks like `hasArguments` is replaced by `hasArgs`. 
So the code should now look like the following - right?

```javascript
import {
    GModelElement,
    hasArgs
} from '@eclipse-glsp/client';
.....

       let element: GModelElement | undefined;
       ....
        if (element && hasArgs(element)) {
              element.args.....
        }
```

**Reply by tortmayr** [reply]
Yes your are correct. We missed this change in the changelog. The feature for adding arbitary arguments has been refactored in 2.0 and is now applied to all model elements per default. In the process the `hasArguments` function was also renamed to `hasArgs`. Sorry, we missed to add this change to the changelog.
